packages/api/AlvTime.Persistence/Repositories/AlvEconomyData/EmployeeHourlySalaryStorage.cs

```diff
@@ -0,0 +1,92 @@
+                ;
+            if (!employeeWithSalaryData.Any())
+                throw new ValidationException("Could not find a registered salary for this user");
+
+            var salary = employeeWithSalaryData.FirstOrDefault(x =>
+                x.FromDateInclusive <= date && (x.ToDate.HasValue && x.ToDate > date || x.ToDate == null));
+
+            return salary.HourlySalary;
+        }
+
+        public void RegisterHourlySalary(EmployeeSalaryDto employeeSalaryData)
+        {
+            var employee = _context.User.FirstOrDefault(e => e.Id == employeeSalaryData.usiderId);
+
+            if (employee == null)
+                throw new ValidationException($"Could not find a an employee with id {employeeSalaryData.usiderId}");
+
+            var employeeHasRegisteredSalary =
+                _economyContext.EmployeeHourlySalaries.Any(x => x.UserId == employeeSalaryData.usiderId);
+
+            if (employeeHasRegisteredSalary)
+            {
+                var currentSalary = _economyContext
+                    .EmployeeHourlySalaries.FirstOrDefault(x => x.UserId == employeeSalaryData.usiderId &&
+                                                                x.FromDateInclusive <= DateTime.Now &&
+                                                                (x.ToDate.HasValue && x.ToDate > DateTime.Now ||
+                                                                 x.ToDate == null));
+                currentSalary.ToDate = employeeSalaryData.FromDate.AddDays(-1);
+            }
+
+            _economyContext.EmployeeHourlySalaries.Add(new EmployeeHourlySalary
+            {
+                UserId = employeeSalaryData.usiderId,
+                FromDateInclusive = employeeSalaryData.FromDate,
+                HourlySalary = employeeSalaryData.HourlySalary,
+                ToDate = employeeSalaryData.ToDate
+            });
+
+
+            _economyContext.SaveChanges();
+        }
+
+        public List<EmployeeSalaryDto> GetEmployeeSalaryData(int userId)
+        {
+            var employeeWithSalaryData = _economyContext
+                    .EmployeeHourlySalaries.Where(hs => hs.UserId == userId)
+                ;
+            if (!employeeWithSalaryData.Any())
+                throw new ValidationException("Could not find a registered salary for this user");
+
+            var returnSalary = new List<EmployeeSalaryDto>();
+            foreach (var hourlySalary in employeeWithSalaryData) returnSalary.Add(ToEmployeeSalaryDto(hourlySalary));
+
+            return returnSalary;
+        }
+
+        private EmployeeSalaryDto ToEmployeeSalaryDto(EmployeeHourlySalary employeeHourlySalary)
+        {
+            return new()
+            {
+                usiderId = employeeHourlySalary.UserId, FromDate = employeeHourlySalary.FromDateInclusive,
+                ToDate = employeeHourlySalary.ToDate, HourlySalary = employeeHourlySalary.HourlySalary
```

Comment:
Vi har ikke vært helt konsekvent på det, men det kan være penere (for frontend) å returnere en dato som string som ikke inneholder tidspunkt. Altså kjøre .ToString("yyyy-MM-dd", CultureInfo.InvariantCulture) etter datoen.

Reply:
Ah, okey. Jeg bruker samme objektet som input i POST EmployeeSalary. Sender frontend da inn dato som string også? Burde jeg opprette eget objekt for input og eget for retur? (Kanskje se litt på den også sammen?)

Reply:
Frontend sender dato inn som dato. Enesten fordelen med å returnere det ut til frontend som string er at da slipper frontend å gjøre noe logikk for å formatere det og kan bare presentere responsen as is. Kan godt hende det er en gode ide å lage to objekter. Vi kan godt se på det sammen i morgen.

Reply:
Okey, jeg laget det om til to ulike records (en for request og en for respons) som vi diskuterte, meeeen jeg gjordemuligens noe rart. I request, trenger jeg på en tidspunkt ID'en, men jeg vil ikke åpne for å ta det som input når du skal registrere ny lønn, og jeg kan ikke bruke responsobjektet fordi det har dato som string. Jeg innførte derfor en tredje record som har med seg Id i tillegg til info fra requesten. Kan du se på det om det ble veldig forvirrende?

![image](https://user-images.githubusercontent.com/79319625/130633050-120335e0-acf8-4517-982f-ae1545233c10.png)

Reply:
Ref kommentar på EmployeeSalary: renamer til EmployeeSalaryDto og fjernet arv da dette var økte lesbarheten